src/components/TextInput/TextInput.tsx

```diff
@@ -42,28 +44,37 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
         name,
         value,
         defaultValue,
+        label,
         disabled = false,
         multiline = false,
         hasClear = false,
         error,
         autoComplete,
         onUpdate,
         onChange,
-        id,
+        id: originalId,
         tabIndex,
         style,
         className,
         qa,
+        controlProps: originalControlProps,
+        labelProps,
     } = props;
     const [uncontrolledValue, setUncontrolledValue] = React.useState(defaultValue ?? '');
     const innerControlRef = React.useRef<HTMLTextAreaElement | HTMLInputElement>(null);
+    const labelRef = React.useRef<HTMLLabelElement>(null);
+    const innerId = useUniqId();
+    const id = label ? originalId || innerId : originalId;
```

Comment:
no need to check label, there is nothing wrong with always setting id to the input

Reply:
ok, will remove (the idea was to avoid changing current behavior for inputs without explicit id provided, but it should be no problem, i guess)